ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/imageCveSingle.test.js

```diff
@@ -0,0 +1,173 @@
+import withAuth from '../../../helpers/basicAuth';
+import { visitWorkloadCveOverview } from './WorkloadCves.helpers';
+import { selectors } from './WorkloadCves.selectors';
+
+describe('Image CVE Single page', () => {
+    withAuth();
+
+    it('should correctly handle ImageCVE single page specific behavior', () => {
+        // Apply global default filters
+        visitWorkloadCveOverview();
+
+        // Click any CVE link in the table to visit the ImageCVE Page.
+        cy.get('tbody tr td a:contains("CVE")').first().click();
+
+        // Wait for the summary card skeleton to disappear
+        cy.get('.pf-c-skeleton:contains("Loading image cve summary data")').should('not.exist');
+
+        // Check that only applicable resource menu items are present in the toolbar
+        cy.get(selectors.resourceDropdown).click();
+        cy.get(selectors.resourceMenuItem('CVE')).should('not.exist');
+        cy.get(selectors.resourceMenuItem('Image'));
+        cy.get(selectors.resourceMenuItem('Deployment'));
+        cy.get(selectors.resourceMenuItem('Cluster'));
+        cy.get(selectors.resourceMenuItem('Namespace'));
+        cy.get(selectors.resourceDropdown).click();
+
+        // The Images toggle button should be the default and the count of the images in the button
+        // should match the count of the images in the "Affected images" card
+        cy.get('button[aria-pressed="true"]:contains("Image")').then((button) => {
+            const buttonImageCount = button.text().replace(/ Image.*/, '');
+
+            cy.get(
+                `${selectors.summaryCard('Affected images')} *:contains("images affected")`
+            ).then((affectedImages) => {
+                const affectedImageCount = affectedImages
+                    .text()
+                    .replace(/\/\d* images affected/, '');
+
+                expect(buttonImageCount).to.equal(affectedImageCount);
```

Comment:
At your convenience, let’s have a call to discuss reliable patterns for quantitative assertions.
* If they are both on the same page and based on the same response, it works to assert consistency.
* If they are on different pages and based on different responses, then ocp-X-Y-ui-e2e-tests have been failing because of changes in number of deployments. Might that also cause different in number of images?

Reply:
Sounds good. I think this case is somewhere in the middle: both are on the same page but based on _different_ responses. This is intended as a consistency check because the two values should always match, and the test failing here would be an indication that something isn't working correctly.

Or so I planned, worth discussing it though since I haven't done much e2e testing since the ocp jobs have been implemented.

Reply:
Removed this to avoid issues with inconsistent responses.